seed/src/lib.rs

```diff
@@ -440,18 +443,25 @@ impl Node {
                 let fetcher = fetcher::PeerToPeer::new(urn.clone(), peer_id, addr_hints)
                     .build(storage)
                     .map_err(|e| Error::MkFetcher(e.into()))??;
-                replication::replicate(storage, fetcher, cfg, None)?;
-                tracking::track(storage, &urn, peer_id)?;
+                let was_updated = tracking::track(storage, &urn, peer_id)?;
+                // Skip explicit replication if we already track the peer --
+                // normal gossip should take care of that.
+                if was_updated {
+                    replication::replicate(storage, fetcher, cfg, None)?;
+                }
 
-                Ok::<_, Error>(())
+                Ok::<_, Error>(was_updated)
             })
             .await?
         };
 
         match &result {
-            Ok(()) => {
+            Ok(true) => {
                 tracing::info!("Successfully tracked project {} from peer {}", urn, peer_id);
             },
+            Ok(false) => {
+                tracing::info!("Project {} from peer {} already tracked", urn, peer_id);
```

Comment:
Not sure, maybe remain silent?

Reply:
Or `debug` if we want to look into this specific call path for some reason